nixos/modules/services/continuous-integration/gitea-actions-runner.nix

```diff
@@ -0,0 +1,175 @@
+{ config
+, lib
+, pkgs
+, utils
+, ...
+}:
+
+let
+  inherit (lib)
+    any
+    attrValues
+    concatStringsSep
+    escapeShellArg
+    hasInfix
+    optionalAttrs
+    optionals
+    literalExpression
+    mapAttrs'
+    mkEnableOption
+    mkOption
+    mkPackageOptionMD
+    mkIf
+    nameValuePair
+    types
+  ;
+
+  inherit (utils)
+    escapeSystemdPath
+  ;
+
+  cfg = config.services.gitea-actions-runner;
+
+  usesContainers = any (instance:
+    # the default label set includes ubuntu containers
+    instance.labels == [] ||
+    # explicit container usage
+    (any (label: hasInfix label "docker://") instance.labels));
+
+  usesDocker = usesContainers && config.virtualisation.docker.enable;
+  usesPodman = usesContainers && config.virtualisation.podman.enable;
```

Comment:
Taking the labels into account now. The default labels that are used, when none are passed, are:
- ubuntu-latest
- ubuntu-22.04
- ubuntu-20.04
- ubuntu-18.04

Reply:
The code block has been updated in the meantime, ptal.